File: compatibility-kit/ruby/features/hooks/hooks.feature.rb
def attach_or_embed(world, data, media_type)
  # Backward compatibility as the steps are also used by cucumber-ruby 3 which does not support `attach`
  world.respond_to?(:attach) ? attach(data, media_type) : embed(data, media_type)
end

Before do
  # no-op
end

When('a step passes') do
  # no-op
end

When('a step throws an exception') do
  raise StandardError, 'Exception in step'
end

After do
  raise StandardError, 'Exception in hook'
end

After('@some-tag or @some-other-tag') do
  raise StandardError, 'Exception in conditional hook'
end

After('@with-attachment') do
  attach_or_embed(self, File.open("#{__dir__}/cucumber.svg"), 'image/svg+xml')
end

File: compatibility-kit/javascript/features/hooks/hooks.feature.ts
import { When, Before, After } from '@cucumber/fake-cucumber'
import fs from 'fs'

Before(function () {
  // no-op
})

When('a step passes', function () {
  // no-op
})

When('a step throws an exception', function () {
  throw new Error('Exception in step')
})

After(function () {
  throw new Error('Exception in hook')
})

After('@some-tag or @some-other-tag', function () {
  throw new Error('Exception in conditional hook')
})

After('@with-attachment', async function () {
  await this.attach(fs.createReadStream(__dirname + '/cucumber.svg'), 'image/svg+xml')
})
